### PR TITLE
Add a new +clanglibcpp option for Boost

### PR DESCRIPTION
**Pull request by daveh86**
Currently, the compile of boost with clang will use the stdlibc++. This patch adds an optional flag to use clangs included libc++ instead.

### CODE REVIEW

File: var/spack/repos/builtin/packages/boost/package.py [hunk 1]
```diff
@@ -288,6 +290,10 @@ def determine_b2_options(self, spec, options):
         #   https://svn.boost.org/trac/boost/ticket/12496
         if spec.satisfies('%clang'):
             options.extend(['pch=off'])
+            if '+clanglibcpp' in spec:
+                options.extend(['toolset=clang',
+                                'cxxflags="-stdlib=libc++"',
```
Comment (junghans) [inline var/spack/repos/builtin/packages/boost/package.py]: Would that overwrite user defined `CFLAGS`?

Comment (daveh86) [reply var/spack/repos/builtin/packages/boost/package.py]: The options here are added to the arguments of b2 (or bjam). They won't externally overload an environment variable. I would need to dive deeper to see what boost does with both arguments and environment variables internally.

**Comment by daveh86** [comment]
autobuild restest this please

**Comment by daveh86** [comment]
It looks like the autobuild needs to be re-run. Can someone internal to the spack team perform this?

**Pull request merged**